test/e2e/providers/cloudprovider.go

```diff
@@ -28,7 +29,7 @@ type CloudProvider interface {
 }
 
 // NewCloudProvider returns a CloudProvider interface or an error
-func NewCloudProvider() (CloudProvider, error) {
+func NewCloudProvider(windowsServerVersion windows.ServerVersion) (CloudProvider, error) {
```

Comment:
I'm not sure if it makes the most sense to have `windowsServerVersion` be defined at CloudProvider struct creation.

What about a case where we want to spin up both a 2019 and 2022 VM? This would prevent that.
Adding the version as a parameter to `GenerateMachineSet()` gives the interface more flexibility.

Reply:
I thought about it more and agree that `windowsServerVersion` should not be the `CloudProvider` struct. However introducing a mixed Windows Server version cluster in CI is going to way more complex and even laying the ground work beyond updating `GenerateMachineSet()` is beyond the scope of this PR.